parts/dcos/bstrap/bootstrapprovision.sh

```diff
@@ -7,11 +7,20 @@ retrycmd_if_failure() { retries=$1; wait=$2; timeout=$3; shift && shift && shift
 TMPDIR="/tmp/dcos"
 mkdir -p $TMPDIR
 
-curl -fLsSv --retry 20 -Y 100000 -y 60 -o $TMPDIR/key https://download.docker.com/linux/ubuntu/gpg &
+# default dc/os component download address (Azure CDN)
+LIBLTDL_DOWNLOAD_URL=https://dcos-mirror.azureedge.net/pkg/libltdl7_2.4.6-0.1_amd64.deb
+DOCKER_CE_DOWNLOAD_URL=https://dcos-mirror.azureedge.net/pkg/docker-ce_17.09.0~ce-0~ubuntu_amd64.deb
+
+case $DCOS_ENVIRONMENT in
+    # because of Chinese GreatWall Firewall, the default packages on Azure CDN is blocked. So the following Chinese local mirror url should be used instead.
+    AzureChinaCloud)
+        LIBLTDL_DOWNLOAD_URL=http://acsengine.blob.core.chinacloudapi.cn/dcos/libltdl7_2.4.6-0.1_amd64.deb
+        DOCKER_CE_DOWNLOAD_URL=http://acsengine.blob.core.chinacloudapi.cn/dcos/docker-ce_17.09.0~ce-0~ubuntu_amd64.deb

```

Comment:
You may use following link for docker_ce package download in China cloud:

http://mirror.kaiyuanshe.cn/docker-ce/linux/ubuntu/dists/xenial/pool/stable/amd64/docker-ce_17.09.0~ce-0~ubuntu_amd64.deb
http://mirror.kaiyuanshe.cn/docker-ce/linux/ubuntu/dists/trusty/pool/stable/amd64/docker-ce_17.09.0~ce-0~ubuntu_amd64.deb
http://mirror.kaiyuanshe.cn/docker-ce/linux/ubuntu/dists/zesty/pool/stable/amd64/docker-ce_17.09.0~ce-0~ubuntu_amd64.deb